### PR TITLE
Added get_best_move_time()

### PR DESCRIPTION
**Pull request by 2kofawsome**
This allows you to enter the time you want stockfish to run for, instead of a predetermined depth (may also help be an alternate for that issue opened about Minimum Thinking Time)

### CODE REVIEW

**Comment by 2kofawsome** [comment]
Why does this say merge 11 commits? (there should be only 1).

I do not have much experience with Github, what did I do wrong?

**Comment by zhelyabuzhsky** [comment]
Thank you for your MR =)

> Why does this say merge 11 commits? (there should be only 1)

no ideas

File: stockfish/models.py [hunk 1]
```diff
@@ -185,6 +188,24 @@ def get_best_move(self) -> Optional[str]:
                 return splitted_text[1]
             last_text = text
 
+    def get_best_move_time(self, time) -> Optional[str]:
```
Comment (zhelyabuzhsky) [inline stockfish/models.py:191]:
```suggestion
    def get_best_move_time(self, time: int) -> Optional[str]:
```

File: stockfish/models.py [hunk 2]
```diff
@@ -85,6 +85,9 @@ def _is_ready(self) -> None:
     def _go(self) -> None:
         self._put(f"go depth {self.depth}")
 
+    def _go_time(self, time) -> None:
```
Comment (zhelyabuzhsky) [inline stockfish/models.py:88]:
```suggestion
    def _go_time(self, time: int) -> None:
```

File: README.md [hunk 3]
```diff
@@ -69,6 +69,16 @@ stockfish.get_best_move()
 d2d4
 ```
 
+### Get best move based on a time constraint
+```python
+stockfish.get_best_move_time('1000')
```
Comment (zhelyabuzhsky) [inline README.md:74]:
```suggestion
stockfish.get_best_move_time(1000)
```

**Comment by 2kofawsome** [comment]
I have made your suggested changes

**Comment by GabrieleMaurina** [comment]
Hi, I did some experiments and the timing aspect seems to work perfectly. Nice job.

So I guess this overrides the minimum thinking time parameter? Does it also override the depth?

**Comment by 2kofawsome** [comment]
Yes it does override the depth

**Pull request merged**